ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright 2020 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.account;
+
+import com.github.ambry.account.mysql.MySqlConfig;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.ambry.account.AccountUtils.*;
+
+
+/**
+ * An implementation of {@link AccountService} that employs MySql database as its underlying storage.
+ */
+public class MySqlAccountService implements AccountService {
+
+  private MySqlAccountStore mySqlAccountStore = null;
+  private final AccountServiceMetrics accountServiceMetrics;
+  private final MySqlConfig mySqlConfig;
+  // in-memory cache for storing account and container metadata
+  private final AccountInfoMap accountInfoMap;
+  private static final Logger logger = LoggerFactory.getLogger(MySqlAccountService.class);
+  private final ReadWriteLock rwLock = new ReentrantReadWriteLock();
+
+  public MySqlAccountService(AccountServiceMetrics accountServiceMetrics, MySqlConfig mySqlConfig) {
+    this.accountServiceMetrics = accountServiceMetrics;
+    this.mySqlConfig = mySqlConfig;
+    accountInfoMap = new AccountInfoMap(accountServiceMetrics);
+    try {
+      createMySqlAccountStore();
+    } catch (SQLException e) {
+      logger.error("MySQL account store creation failed", e);
```

Comment:
This is okay for now.  We will need to distinguish between transient connection error and something like credential issue which should fail startup.

Reply:
Sure, added a todo here.